### PR TITLE
feat: support placeholder queries that only request a subset of data

### PR DESCRIPTION
**Pull request by WinPlay02**
- changed `placeholder_query` messages (VSCode to Runner) to allow optional `window_begin` and `window_size` fields (in addition to the placeholder `name`)
- changed `placeholder_value` messages (Runner to VSCode) to include `window_begin`, `window_size` and `window_max` fields and `windowed` flag when querying only windows of data

### CODE REVIEW

File: packages/safe-ds-vscode/src/extension/messages.ts [hunk 1]
```diff
@@ -68,7 +68,17 @@ export interface ProgramMainInformation {
 export interface PlaceholderQueryMessage {
     type: 'placeholder_query';
     id: string;
-    data: string;
+    data: PlaceholderQuery;
+}
+
+/**
+ * Contains the query for a placeholder value.
+ * Field name contains the name of the requested placeholder, field window_begin may contain an offset of and window_size may contain the size of a subset of the requested data.
```
Comment (lars-reimann) [inline packages/safe-ds-vscode/src/extension/messages.ts:76]: Better add a `/** ... */` comment to the individual fields.

File: packages/safe-ds-vscode/src/extension/messages.ts [hunk 2]
```diff
@@ -68,7 +68,27 @@ export interface ProgramMainInformation {
...
+
+    /**
+     * The offset of the requested data.
+     */
+    window_begin?: number;
+
+    /**
+     * The size of the requested data.
+     */
+    window_size?: number;
```
Comment (lars-reimann) [inline packages/safe-ds-vscode/src/extension/messages.ts:91]: Is it intended that you can also specify only `window_begin` or `window_size`? What's the behavior in this case? I guess `window_begin` defaults to the start of the data and `window_size` to the size of the remaining data?

File: packages/safe-ds-vscode/src/extension/messages.ts [hunk 3]
```diff
@@ -105,6 +125,10 @@ export interface PlaceholderValue {
     name: string;
     type: string;
     value: string;
+    windowed?: boolean;
+    window_begin?: number;
+    window_size?: number;
+    window_max?: number;
```
Comment (lars-reimann) [inline packages/safe-ds-vscode/src/extension/messages.ts:131]: I have a similar question here. Though the combination `windowed: false` and `window_size: 50` doesn't seem to make much sense either way. It seems like in this case, either all fields must be set or none. Thus, I'd propose to add some nesting here:

```ts
{
    // ...
    value: string;
    window?: {
        begin: number;
        size: number;
        max: number;
    }
}
```

For consistency, we should then also use this nested structure for the `PlaceholderQuery`.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: packages/safe-ds-vscode/src/extension/messages.ts (hunk 2 above)
Comment (WinPlay02) [reply packages/safe-ds-vscode/src/extension/messages.ts:91]: That is correct

File: packages/safe-ds-vscode/src/extension/messages.ts (hunk 1 above)
Comment (WinPlay02) [reply packages/safe-ds-vscode/src/extension/messages.ts]: Where applicable, comments for other messages were also restructured.

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**